futures-util/src/future/shared.rs

```diff
@@ -93,22 +93,31 @@ where
     Fut: Future,
     Fut::Output: Clone,
 {
-    /// If any clone of this `Shared` has completed execution, returns its result immediately
+    /// If any clone of this `Shared` has completed execution and there are still at least one
+    /// clone of this `Shared` alive then, returns its result immediately
     /// without blocking. Otherwise, returns None without triggering the work represented by
     /// this `Shared`.
```

Comment:
I think this text doesn't match the current implementation. Here's what I suggest:

> Returns `Some` containing a reference to this `Shared`'s output if it has already been computed by a clone or `None` if it hasn't been computed yet or if this `Shared` already returned its output from `poll`.

(Also, this text break at 80 chars per line)